domain/consensus/processes/blockprocessor/validateandinsertpruningpoint_test.go

```diff
@@ -0,0 +1,164 @@
+		if err != nil {
+			t.Fatalf("Error setting up tcSyncer: %+v", err)
+		}
+		defer teardownSyncer(false)
+
+		tcSyncee, teardownSyncee, err := factory.NewTestConsensus(params, false, "TestValidateAndInsertPruningPointSyncee")
+		if err != nil {
+			t.Fatalf("Error setting up tcSyncee: %+v", err)
+		}
+		defer teardownSyncee(false)
+
+		addBlock := func(parentHashes []*externalapi.DomainHash) *externalapi.DomainHash {
+			block, _, err := tcSyncer.BuildBlockWithParents(parentHashes, nil, nil)
+			if err != nil {
+				t.Fatalf("BuildBlockWithParents: %+v", err)
+			}
+
+			_, err = tcSyncer.ValidateAndInsertBlock(block)
+			if err != nil {
+				t.Fatalf("ValidateAndInsertBlock: %+v", err)
+			}
+
+			_, err = tcSyncee.ValidateAndInsertBlock(&externalapi.DomainBlock{
+				Header:       block.Header,
+				Transactions: nil,
+			})
+			if err != nil {
+				t.Fatalf("ValidateAndInsertBlock: %+v", err)
+			}
+
+			return consensushashing.BlockHash(block)
+		}
+
+		tipHash := params.GenesisHash
+		for i := 0; i < finalityDepth-2; i++ {
+			tipHash = addBlock([]*externalapi.DomainHash{tipHash})
+		}
+
+		// Add block in the anticone of the pruning point to test such situation
+		pruningPointAnticoneBlock := addBlock([]*externalapi.DomainHash{tipHash})
+		tipHash = addBlock([]*externalapi.DomainHash{tipHash})
+		nextPruningPoint := addBlock([]*externalapi.DomainHash{tipHash})
+
+		tipHash = addBlock([]*externalapi.DomainHash{pruningPointAnticoneBlock, nextPruningPoint})
+
+		// Add blocks until the pruning point changes
+		for {
+			tipHash = addBlock([]*externalapi.DomainHash{tipHash})
+
+			pruningPoint, err := tcSyncer.PruningPoint()
+			if err != nil {
+				t.Fatalf("PruningPoint: %+v", err)
+			}
+
+			if !pruningPoint.Equal(params.GenesisHash) {
+				break
+			}
+		}
+
+		pruningPoint, err := tcSyncer.PruningPoint()
+		if err != nil {
+			t.Fatalf("PruningPoint: %+v", err)
+		}
+
+		if !pruningPoint.Equal(nextPruningPoint) {
+			t.Fatalf("Unexpected pruning point %s", pruningPoint)
+		}
+
+		pruningPointUTXOSet, err := tcSyncer.GetPruningPointUTXOSet(pruningPoint)
+		if err != nil {
+			t.Fatalf("GetPruningPointUTXOSet: %+v", err)
+		}
+
+		tip, err := tcSyncer.GetBlock(tipHash)
+		if err != nil {
+			t.Fatalf("GetBlock: %+v", err)
+		}
+
+		// Check that ValidateAndInsertPruningPoint fails for invalid pruning point
+		err = tcSyncee.ValidateAndInsertPruningPoint(tip, pruningPointUTXOSet)
+		if !errors.Is(err, ruleerrors.ErrUnexpectedPruningPoint) {
+			t.Fatalf("Unexpected error: %+v", err)
+		}
+
+		pruningPointBlock, err := tcSyncer.GetBlock(pruningPoint)
+		if err != nil {
+			t.Fatalf("GetBlock: %+v", err)
+		}
+
+		invalidPruningPointBlock := pruningPointBlock.Clone()
+		invalidPruningPointBlock.Transactions[0].Version += 1
+
+		// Check that ValidateAndInsertPruningPoint fails for invalid block
+		err = tcSyncee.ValidateAndInsertPruningPoint(invalidPruningPointBlock, pruningPointUTXOSet)
```

Comment:
Maybe also try to mutate `pruningPointUTXOSet` and see that it fails?
I'd check both naive mutation, and actually changing one of the UTXOs inside